torch/csrc/jit/pickler.h

```diff
@@ -183,19 +185,40 @@ class Pickler {
 
   const void* getPointer(const IValue& ivalue);
 
+  // Caller checks that bufferPos_ > 0
+  void flushNonEmpty() {
+    writer_(buffer_.data(), bufferPos_);
+    bufferPos_ = 0;
+  }
+
+  void flush() {
+    if (bufferPos_ != 0) {
+      flushNonEmpty();
+    }
+  }
+
   // These convert values to bytes and add them to the stack (NB: since T is to
   // the left of a '::', its type cannot be deduced by the compiler so one must
   // explicitly instantiate the template, i.e. push<int>(int) works, push(int)
   // does not)
   template <typename T>
   void push(typename std::common_type<T>::type value) {
     const char* begin = reinterpret_cast<const char*>(&value);
-    writer_(begin, sizeof(T));
+    if (bufferPos_ + sizeof(T) > buffer_.size()) {
```

Comment:
we should static assert that `sizeof(T) < 256`, otherwise the memcpy is invalid. This seems unlikely to happen, but it is possible if someone changes value to no longer be common_type, and then passes in a large struct.

Reply:
Good idea. I was half-thinking about this at the time, but it didn't make it into final diff, uploading a new version with this.